### PR TITLE
Add the DotnetPlatform package type

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/7337
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
Adding the `DotnetPlatform` package type. This package type is not compatible in the regular projects.

## Testing/Validation

Tests Added: Yes  
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/Core/PackageType.cs [hunk 1]
```diff
@@ -17,6 +17,9 @@ public class PackageType : IEquatable<PackageType>, IComparable<PackageType>
         public static readonly PackageType Dependency = new PackageType("Dependency", version: EmptyVersion);
         public static readonly PackageType DotnetTool = new PackageType("DotnetTool", version: EmptyVersion);
         public static readonly PackageType SymbolsPackage = new PackageType("SymbolsPackage", version: EmptyVersion);
+        public static readonly PackageType DotnetPlatform = new PackageType("DotnetPlatform", version: EmptyVersion);
+
+        public static StringComparer PackageTypeNameComparer = StringComparer.OrdinalIgnoreCase;
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Core/PackageType.cs]: `readonly`

File: src/NuGet.Core/NuGet.Packaging/Core/PackageType.cs [hunk 2]
```diff
@@ -113,5 +111,10 @@ public int CompareTo(PackageType other)
 
             return Version.CompareTo(other.Version);
         }
+
+        public override string ToString()
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Core/PackageType.cs]: Why is this necessary?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Core/PackageType.cs]: It's not anymore. In an earlier iteration I was doing a ToString in the log message, but I changed my mind. I will remove it.

**Review by dtivel — APPROVED** [review]
